File: Tareas/13/Código RUBY/tarea13.rb
=begin
	Nombre: Leslie Estefanía Pérez Cerda
	Matrícula: 1737535
	Hora: Jueves V4-V6
=end

print("Ingrese el numero de terminos: ")
n = gets().to_i

serie = ""
x = 0
y = 2
z = 1
a = 0
b = -3
c = 2

if(n>0)then
	serie = serie + "2"

	for i in(1...n)
		serie = serie + ", "

		if(z==0)then
			c = c + 2
			serie = serie + c.to_s

			if(x<a)then
				x = x + 1
			else
				z = 1
				x = 0
			end
		elsif(z==1)then
			serie = serie + b.to_s
			b = b - 2
			if(x<y)then
				x = x + 1
			else
				z = 0
				x = 0
				a = a + 1
			end
		end
	end
end

print(serie)

File: Tareas/13/Código JAVA/tarea13.java
/*	
	Nombre: Leslie Estefanía Pérez Cerda
	Matrícula: 1737535
	Hora: Jueves V4-V6
*/

import javax.swing.JOptionPane;
public class tarea13{
	public static void main (String[] args){
		String entrada;
		String serie = "";
		int n,i,x=0,y=2,z=1,a=0,b=-3,c=2;

		entrada = JOptionPane.showInputDialog("Ingrese el numero de terminos: ");
		n = Integer.parseInt(entrada);

		if(n>0){
			serie = serie + "2";

			for(i=1; i<n; i++){
				serie = serie + ", ";
				if(z==0){
					c = c + 2;
					serie = serie + c;

					if(x<a){
						x++;
					}else{
						z = 1;
						x = 0;
					}
				}else if(z==1){
					serie = serie + b;
					b = b - 2;
					if(x<y){
						x++;
					}else{
						z = 0;
						x = 0;
						a++;
					}
				}
			}
		}
		JOptionPane.showMessageDialog(null,serie);
	} //Fin de función main
} //Fin de clase tarea13